mcsema/Arch/ABI.cpp

```diff
@@ -524,10 +527,76 @@ llvm::Value *CallingConvention::LoadNextArgument(llvm::BasicBlock *block,
   // Bump the stack pointer.
   alloc_size = std::max<uint64_t>(alloc_size, addr_size);
   num_loaded_stack_bytes += alloc_size;
-
   return val;
 }
 
+
+llvm::Value *CallingConvention::LoadNextArgument(llvm::BasicBlock *block,
+                                                 llvm::Type *target_type,
+                                                 bool isByVal) {
+  if (!target_type) {
+    target_type = gWordType;
+  }
+
+  llvm::IRBuilder<> ir(block);
+
+  std::vector<llvm::Value*> underlying_values;
+  llvm::Type* goal_type = target_type;
+
+  if (target_type->isPointerTy() && !isByVal) {
+    return LoadNextSimpleArgument(block, target_type);
+  } else if (auto ptr_type = llvm::dyn_cast<llvm::PointerType>(target_type)){
+    goal_type = ptr_type->getElementType();
+  }
+  if (auto vector_type = llvm::dyn_cast<llvm::VectorType>(goal_type)) {
+
+    llvm::Type* under_type = vector_type->getElementType();
+    for (uint64_t i = 0; i < vector_type->getNumElements(); ++i) {
+      underlying_values.push_back(LoadNextSimpleArgument(block, under_type));
+    }
+    llvm::Value* last_val = llvm::Constant::getNullValue(goal_type);
+
+    for (uint64_t i = 0; i < underlying_values.size(); ++i) {
+      last_val = ir.CreateInsertElement(last_val, underlying_values[i],
+          llvm::ConstantInt::get(llvm::Type::getInt64Ty(*gContext), i));
```

Comment:
Move `last_val, ...` to next line as well.

Reply:
Maybe hoist out the call `llvm::Type::getInt64Ty(*gContext)` into an `auto int64_type = llvm::Type::getInt64Ty(*gContext);` outside of the loop.